### DISCUSSION TITLE
Types for UseFormRegister in subcomponent

### DISCUSSION THREAD

**Discussion by simonpweller** [react-hook-form/react-hook-form#5355] [Q&A] [answered]
I am trying to understand whether/how I can extract parts of a form into a separate component, so it can be used across multiple forms. 

To illustrate, I made this [example codesandbox](https://codesandbox.io/s/react-hook-form-ts-subcomponent-brify?file=/src/index.tsx)

In the subcomponent, I would like to hide the unrelated fields of the parent form since those may differ based on the context, so i have only included the relevant fields in the type of the register prop. 

However, this leads to a type error in the parent prop: `Property 'age' is missing in type 'INameInput' but required in type 'IFormInput'.ts`

How can I fix this? Is this the wrong approach altogether?

**Comment by simonpweller** [comment, marked as answer]
I managed to sort of solve this using Partial [updated example here](https://codesandbox.io/s/react-hook-form-ts-subcomponent-v2-bymg2). 

Unfortunately I can't see a way how I could re-use the same subform twice in a parent form this way (for example: using an address form to record a billing and a shipping address). Ideally, I would like to be able to pass in a parentName prop to the subform like in the example [here](https://github.com/nareshbhatia/form-examples/blob/main/checkout-form-rhf/src/components/AddressForm/AddressForm.tsx#L9), but I can't seem to find a way to do this without losing type safety.

Even if I [enumerate the possible prefixes](https://codesandbox.io/s/react-hook-form-ts-subcomponent-v3-ufffb?file=/src/index.tsx), TypeScript doesn't see how the signatures would be compatible with one another.

**Reply by bluebill1049** [reply]
have you considered pass only the methods down? instead of the register method?

**Reply by simonpweller** [reply]
Yeah that works, thanks. I have updated my original example and I'm pretty happy with [the result](https://codesandbox.io/s/react-hook-form-ts-subcomponent-brify?file=/src/index.tsx)- thanks for responding.
